packages/x-data-grid/src/utils/cleanupTracking/TimerBasedCleanupTracking.ts

```diff
@@ -14,7 +14,7 @@ export class TimerBasedCleanupTracking implements CleanupTracking {
 
   register(object: any, unsubscribe: UnsubscribeFn, unregisterToken: UnregisterToken): void {
     if (!this.timeouts) {
-      this.timeouts = new Map<number, NodeJS.Timeout>();
+      this.timeouts = new Map<number, ReturnType<typeof setTimeout>>();
```

Comment:
Given that we need specificity to have conclusive types, I would vote for reverting https://github.com/mui/mui-public/issues/195 and using `window.setTimeout` with return type of `number` instead of relying on terse `ReturnType<typeof setTimeout>`, which would now return `NodeJS.Timeout` in IDE/JSDoc and `number` in built output.

Reply:
This makes perfect sense ... I was wondering why we use it without the `window` prefix as well at some point

Reply:
window and global/globalThis are different things and can be misleading. I think that is the main argument to not use things from `window` which are usually present in `global`.

Reply:
Why would IDE's keep returning `NodeJS.Timeout`?

Reply:
Because `setTimeout` exists on `Node` and in regular (not build) `tsconfig`, we list the `node` types to make linting happy (we use some node APIs in tests and scripts)

Reply:
I believe your IDE looks at the [tsconfig.json](https://github.com/mui/mui-x/blob/6e5415288acf9098ae78c737e9041b694639b073/packages/x-charts/tsconfig.json#L9) and sees node types. In principle I'd say we should just remove node types everywhere, and write tests that can run both in node and in the browser. but that's probably easier said than done.

Reply:
> I think it should be fine to remove the `node` type from the packages.
> 
> Have you tried that yet?
> 
> We might need to add it to auxiliary folders tsconfig though. Like in `test/tsconfig.json` and probably even in the root config.

I have tried it, but it is unrealistic due to `test/utils` folder.
We would need to refactor to using it as a package and add `skipLibCheck: "true"` to every package to avoid transitive issues from it. 🙈 
We would get similar issues from the `@mui/x-telemetry` package... 🤷

Reply:
in principle for code that runs on different environments you write different tsconfig files. I don't think they necessary need to be separate packages. Our current build tools wouldn't work with it though

Reply:
That is what we have now more than before this PR.
In IDE and typechecking, we use the `tsconfig.json` file, while type building relies on the `tsconfig.build.json` file. 🤔

Reply:
I mean multiple tsconfig files that each include different parts of the code. Like e.g. the vite [example](https://github.com/mui/material-ui/blob/master/examples/material-ui-vite-ts/tsconfig.json).